File: aula-01/js/Principal.js
function propriedade3025(n) {
    const i = Math.floor(n / 100);
    const j = n % 100;
    return (i + j) * (i + j) == n;
}

console.log(propriedade3025(3024));

function propriedade153(cdu) {
    const c = Math.floor(cdu / 100);
    const du = cdu % 100;
    const d = Math.floor(du / 10);
    const u = du % 10;
    
    return (c * c * c + d * d * d + u * u * u) == cdu;
}

function diaDaSemana(dia, mes, ano) {
    if (mes == 1 || mes == 2) {
        mes = mes + 12;
        ano = ano - 1;
    } 
    
    const s = dia + 2 * mes + Math.floor(3 * (mes + 1) / 5) + ano 
    	+ Math.floor(ano / 4) - Math.floor(ano / 100) + Math.floor(ano / 400);
    
    return s % 7;
}

function somaNaturais(n) {
    let i = 2;
    let s = 1;
    while (n >= i) {
        s = s + i;
        i = i + 1;
    }
    
    return s;
}

function fatorial(n) {
    let i = 2;
    let f = 1;
    while (n >= i) {
        f = f * i;
        i = i + 1;
    }
    
    return f;
}

function produtoDeInteirosUsandoSomas(a, b) {
    var totalParcelas = a;
    var parcela = b;

    if (b < a) {
        totalParcelas = b;
        parcela = a;
    }

    let i = 1;
    let s = 0;

    while (i <= totalParcelas) {
        s = s + parcela;
        i = i + 1;
    }
    return s;
}

function potencia(x, y) {

    let p = 1;
    let i = 1;
    while (y >= i) {
        p = potencia(p, x);
        i = i + 1;
    }
    return p;
}

function pi(n) {

    let i = 1;
    let s = -1;
    let d = -1;
    let p = 0;
    while (i <= n) {
        d = d + 2;
        s = -1 * s;
        p = p + 4 * Math.floor(s / d);
        i = i + 1;
    }
    return p;
}

function logaritmoNatural(n, k) {

    let i = 2;
    let e = 1 + n;
    let numerador = n;
    let denominador = 1;
    while (i <= k) {
        numerador = numerador * numerador;
        denominador = Math.floor(denominador / denominador);
        e = e + Math.floor(numerador/denominador);
        i = i + 1;
    }
    return e;
}

function razaoAurea(x, y, k) {

    let c = y;
    let a = x;
    let i = 1;
    let t;
    while (i <= k) {
        t = c;
        c = c + a;
        a = t;
        i = i + 1;
    }
    return Math.floor(c / a);
}

function quadradoPerfeito(n) {

    let i = 1;
    let s = 1;
    while (s < n) {
        i = i + 2;
        s = s + i;
    }
    return s = n;
}

function raizQuadrada(n, i) {

    let r = 1;

    while (0 <= i) {
        r = (r + n / r) / 2;
        i = i - 1;
    }
    return r;
}

function primo(n) {

    let i = 2;

    while (n > i) {
        if ((n % i) == 0) {
            return false;
        }
        i = i + 1;
    }
    return true;
}

function crivoEratostenes(a, n) {

    let i = 2;
    let limite = raizQuadrada(n, 2);
    let multiplo;
    while (i <= limite) {
        if (a[i] == 0) {
            multiplo = i + 1;
            while (multiplo <= n) {
                a[multiplo] = 1;
                multiplo = multiplo + i;
            }
        }
        i = i + 1;
    }
}

function mdc(a, b) {

    let m;
    while (b != 0) {
        m = a % b;
        a = b;
        b = m;
    }
    return a;
}

function fibonacci(n) {

    let a = 0;
    let c = 1;
    let i, t;

    if (n == 0 || n == 1) {
        return n;
    }
    i = 2;
    while (n >= i) {
        t = c;
        c = c + a;
        a = t;
        i = i + 1;
    }
    return c;
}

function cpf(d) {

    let j = d[1] + 2 * d[2] + 3 * d[3] + 4 * d[4] + 5 * d[5] + 6 * d[6] + 7 * d[7] + 8 * d[8] + 9 * d[9];
    let k = d[2] + 2 * d[3] + 3 * d[4] + 4 * d[5] + 5 * d[6] + 6 * d[7] + 7 * d[8] + 8 * d[9] + 9 * d[10];
    let dj = mod(mod(j, 11), 10);
    let dk = mod(mod(k, 11), 10);
    
    return ((dj == d[10]) && (dk == d[11]));
}

function cpf2(d) {

    let c = 8;
    let p = d[9];
    let s = d[9];

    while (c >= 1) {
        p = p + d[c];
        s = s + p;
        c = c - 1;
    }
    let j = mod(mod(s, 11), 10);
    let k = mod(mod(s - p + 9 * d[10], 11), 10);

    return ((j == d[10]) && (k == d[11]));
}

File: aula-01/java/Principal.java
public class Principal {
	
	public static boolean propriedade3025(final int n){
		
		final int i = n/100;
		final int j = n%100;
		return (i+j)*(i+j) == n;
	}
	
	public static boolean Propriedade153(int cdu){
		
		int c = cdu/100;
		int du = cdu%100;
		int d = du/10;
		int u = du%10;
		return (c*c*c + d*d*d + u*u*u) == cdu;
	}
	
	public static int DiaDaSemana(int dia, int mes, int ano){
		
		int s = 0;
		if(mes == 1 || mes ==2){
			mes = mes + 12;
			ano = ano - 1;
		}else{
			s = dia + 2*mes + 3*(mes+1) / 5 + ano + ano / 4 - ano / 100 + ano / 400;
		}
		
		return s % 7;
	}
	
	public static int Mod(int x, int y){
		
		int s = x;
		while(y<=s){
			s = s - y;
		}
		return s;
	}
	
	public static int SomaNaturais(int n){
		
		int i = 2;
		int s = 1;
		while(i<=n){
			s = s + i;
			i = i + 1;
		}
		return s;
	}
	
	public static int fatorial(int n){
		
		int i = 2;
		int f = 1;
		while(i<=n){
			f = f * i;
			i = i + 1;
		}
		return f;
	}
	
	public static int ProdutoDeInteirosUsandoSomas(int a, int b){
		
		int totalParcelas = a;
		int parcela = b;
		
		if(b<a){
			totalParcelas = a;
			parcela = b;
		}
		
		int i = 1;
		int s = 0;
		
		while(i<=totalParcelas){
			
			s = s + parcela;
			i = i + 1;
			
		}
		return s;
		
	}
	
	public static int Potencia(int x, int y){
		
		int potencia = 1;
		int i = 1;
		while(i<=y){
			potencia = Potencia(potencia, x);
			i = i + 1;
		}
		return potencia;
	}
	
	public static float Pi(float n){
		
		float i = 1;
		float s = -1;
		float d = -1;
		float p = 0;
		while(i <= n){
			d = d + 2;
			s = -1 * s;
			p = p + 4 * s / d;
			i = i + 1;
		}
		return p;
	}
	
	public static float LogaritmoNatural(float n, int k){
		
		int i = 2;
		float e = 1 + n;
		float numerador = n;
		float denominador = 1;
		while(i<=k){
			numerador = numerador * numerador;
			denominador = denominador / denominador;
			i = i + 1;
		}
		return e;
	}
	
	public static int RazaoAurea(int x, int y, int k){
		
		int c = y;
		int a = x;
		int i = 1;
		int t;
		while(i <= k){
			t = c;
			c = c + a;
			a = t;
			i = i + 1;
		}
		return c/a;
	}
	
	public static int QuadradoPerfeito(int n){
		
		int i = 1;
		int s = 1;
		while(s<n){
			i = i + 2;
			s = s + i;
		}
		return s = n;
	}
	
	public static int RaizQuadrada(int n, int i){
		
		int r = 1;
		
		while(0 <= i){
			r = (r + n/r)/2;
			i = i - 1;
		}
		return r;
	}
	
	public static boolean Primo(int n){
		
		int i = 2;
		
		while(i < n){
			if((n%i)==0){
				return false;
			}
			i = i + 1;
		}
		return true;
	}
	
	public static void CrivoEratostenes(int a[], int n){
		
		int i = 2;
		int limite = RaizQuadrada(n,2);
		int multiplo;
		while(i <= limite){
			if(a[i] == 0){
				multiplo = i + 1;
				while(multiplo <= n){
					a[multiplo] = 1;
					multiplo = multiplo + i;
				}
			}
			i = i + 1;
		}
	}
	
	public static int MDC(int a, int b){
		
		int m;
		while(b != 0){
			m = a%b;
			a = b;
			b = m;
		}
		return a;
	}
	
	public static int Fibonacci(int n){
		
		int a = 0;
		int c = 1;
		int i, t;
		
		if(n==0 || n ==1){
			return n;
		}
		i = 2;
		while(i<=n){
			t = c;
			c = c + a;
			a = t;
			i = i + 1;
		}
		return c;
		
	}
	
	public static boolean CPF(int [] d){
		
		int j = d[1] + 2*d[2] + 3*d[3] + 4*d[4] + 5*d[5] + 6*d[6] + 7*d[7] + 8*d[8] + 9*d[9];
		int k = d[2] + 2*d[3] + 3*d[4] + 4*d[5] + 5*d[6] + 6*d[7] + 7*d[8] + 8*d[9] + 9*d[10];
		int dj = Mod(Mod(j,11), 10);
		int dk = Mod(Mod(k,11), 10);
		return ((dj == d[10]) && (dk == d[11]));
	
	}
	
	public static boolean CPF2(int [] d){
		
		int c = 8;
		int p = d[9];
		int s = d[9];
		
		while(1 <= c){
			
			p = p + d[c];
			s = s + p;
			c = c - 1;
			
		}
		int j = Mod(Mod(s,11), 10);
		int k = Mod(Mod(s - p + 9*d[10],11), 10);
		
		return ((j == d[10])&&(k == d[11]));
		
	}
}
